cpp/src/distance/cluster/kmeans_init_plus_plus_double.cu

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <raft/cluster/kmeans.cuh>
+#include <raft/core/device_resources.hpp>
+#include <raft/distance/specializations.cuh>
+
+namespace raft::runtime::cluster::kmeans {
+
+void init_plus_plus(raft::device_resources const& handle,
+                    const raft::cluster::kmeans::KMeansParams& params,
+                    raft::device_matrix_view<const double, int> X,
+                    raft::device_matrix_view<double, int> centroids)
+{
+  // XXX what should the size of this be?
+  rmm::device_uvector<char> workspace(10, handle.get_stream());
```

Comment:
What is a good size for this temporary workspace? Does it even matter or is it anyway resized to the size that is needed inside the function(s) that use it?

Reply:
In general we have been working to remove the temporary workspace arguments from our public APIs in favor of using the "workspace_resource" in the device_resources instance. However, we can keep this one the way it is in the meantime until we scrape through and clean it up. The device_uvector is an RAII compliant (owns the memory, can resize it, and frees it when destructed) so you can just pass in a 0 size workspace and it'll get resized accordingly inside the init_plus_plus function.